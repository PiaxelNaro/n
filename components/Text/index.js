import styled from 'styled-components';

const Text = styled.div`
    /* override color */
    color: ${props => (props.color ? props.color : 'inherit')};

    /* text ellipsis */
    word-break: ${props => (props.ellipsis ? 'break-all' : 'inherit')};
    white-space: ${props => (props.ellipsis ? 'nowrap' : 'inherit')};
    text-overflow: ${props => (props.ellipsis ? 'ellipsis' : 'inherit')};
    overflow: ${props => (props.ellipsis ? 'hidden' : 'inherit')};

    /* text transform */
    text-transform: ${props => (props.textTransform ? props.textTransform : 'inherit')};
`;

export default Text;
