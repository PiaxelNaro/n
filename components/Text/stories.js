import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Text from '.';

const lorem =
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';
storiesOf('Text', module).add('Ellipsis Text', () => (
    <div style={{ width: '200px' }}>
        <Text color="red" ellipsis>{lorem}</Text>
    </div>
));
