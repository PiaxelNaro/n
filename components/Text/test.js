// @flow
import React from 'react';
import { mount } from 'enzyme';
import Text from '.';

describe('<Text>', () => {
    it('render', () => {
        const text = mount(<Text />);
        expect(text).toMatchSnapshot();
        expect(text.prop('ellipsis')).toBeUndefined();
        expect(text.prop('size')).toBeUndefined();
        expect(text.prop('color')).toBeUndefined();
        expect(text.prop('width')).toBeUndefined();
    });

    it('render text', () => {
        const text = mount(<Text ellipsis />);
        expect(text).toMatchSnapshot();
        expect(text.prop('ellipsis')).toBeTruthy();

        text.setProps({
            size: '16px',
        });
        text.update();
        expect(text.prop('size')).toEqual('16px');

        text.setProps({
            color: 'red',
        });
        text.update();
        expect(text.prop('color')).toEqual('red');

        text.setProps({
            width: '10px',
        });
        text.update();
        expect(text.prop('width')).toEqual('10px');

        text.setProps({
            textTransform: 'uppercase',
        });
        text.update();
        expect(text.prop('textTransform')).toEqual('uppercase');
    });
});
