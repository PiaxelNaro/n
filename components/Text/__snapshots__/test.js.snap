// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> render 1`] = `
<Text__Text>
  <div
    className="Text__Text-kba93n-0 hzMLqL"
  />
</Text__Text>
`;

exports[`<Text> render text 1`] = `
<Text__Text
  ellipsis={true}
>
  <div
    className="Text__Text-kba93n-0 ipsvzA"
  />
</Text__Text>
`;
